ruby/command-t/scanner/file_scanner.rb

```diff
@@ -81,35 +102,11 @@ def ensure_cache_under_limit
         @paths_keys << @path
       end
 
-      def path_excluded?(path, prefix_len = @prefix_len)
-        if apply_wild_ignore?
-          # first strip common prefix (@path) from path to match VIM's behavior
-          path = path[prefix_len..-1]
-          path = VIM::escape_for_single_quotes path
-          ::VIM::evaluate("empty(expand(fnameescape('#{path}')))").to_i == 1
-        end
-      end
-
-      def has_custom_wild_ignore?
-        !!@wild_ignore
-      end
-
-      # Used to skip expensive calls to `expand()` when there is no applicable
-      # wildignore.
-      def apply_wild_ignore?
-        if has_custom_wild_ignore?
-          !@wild_ignore.empty?
-        else
-          !!@base_wild_ignore
+      def path_excluded?(path)
+        @wild_ignore_setting.any? do |pattern|
+          File::fnmatch? pattern, path
         end
       end
-
-      def set_wild_ignore(&block)
-        ::VIM::command("set wildignore=#{@wild_ignore}") if has_custom_wild_ignore?
-        yield
-      ensure
```

Comment:
Note here that `ensure` is used to perform clean-up to restore a setting to its former state even when the (arbitrary) code executed by the `yield` call raises an exception.